Amended README,POM and represented all the methods on the tests in XML format

pioushasan2305 opened the pull request.

pioushasan2305 committed:
Updated README.md and pom.xml to fix testrunner issue

pioushasan2305 committed:
Listed all methods under the tests and represented as XML file

winglam commented on dt-impact-tracer/src/main/java/edu/washington/cs/dt/impact/util/RuntimeGenerator.java:
```diff
@@ -87,7 +96,7 @@ public static void main(String[] args) throws IOException {
                             File testDir = new File(testOutput+name);
                             Scanner testReader = new Scanner(testDir);
-                            File temp = new File("new_csv.txt");
+                            File temp = new File("new_xml.txt");
```
What is this file for? Can we provide a more descriptive name?

pioushasan2305 replied:
@winglam Its a temporary file to save and later parse through sootTimerOutput data. I will rename this to more descriptive name in next pull request.

Talank commented on dt-impact-tracer/src/main/java/edu/washington/cs/dt/impact/util/RuntimeGenerator.java:
```diff
@@ -65,16 +73,17 @@ public static void main(String[] args) throws IOException {
                         endTest = "endAfter";
                     }
-                    csvPrinter.printRecord(name, startTest, "", timeStamp, "");
+                    createeElemnthead(doc,test,name,timeStamp);
```
```suggestion
                    createElemnthead(doc,test,name,timeStamp);
```

Talank commented on dt-impact-tracer/src/main/java/edu/washington/cs/dt/impact/util/RuntimeGenerator.java:
```diff
@@ -65,16 +73,17 @@ public static void main(String[] args) throws IOException {
                                 String fullMethodName = subReader.nextLine();
                                 String methodNameInfo = fullMethodName.split(" >>> ")[1];
-                                csvPrinter.printRecord(name, part, methodNameInfo.split(" : ")[0].trim(), methodNameInfo.split(" : ")[1].trim(), returnFrom);
+                                createeElemnt(doc,methods,methodNameInfo.split(" : ")[0].trim(), methodNameInfo.split(" : ")[1].trim(), returnFrom);
```
```suggestion
                                createElemnt(doc,methods,methodNameInfo.split(" : ")[0].trim(), methodNameInfo.split(" : ")[1].trim(), returnFrom);
```

Talank commented on dt-impact-tracer/src/main/java/edu/washington/cs/dt/impact/util/RuntimeGenerator.java:
```diff
@@ -98,7 +107,7 @@ public static void main(String[] args) throws IOException {
                                     String subMethod = method_name.split(" >>> ")[1];
                                     if(subMethod.split(" : ")[0].trim().equals(fullMethodName.trim())){
-                                        csvPrinter.printRecord(name, part, subMethod.split(" : ")[0].trim(), subMethod.split(" : ")[1].trim(), returnFrom);
+                                        createeElemnt(doc,methods,subMethod.split(" : ")[0].trim(), subMethod.split(" : ")[1].trim(), returnFrom);
```
```suggestion
                                        createElemnt(doc,methods,subMethod.split(" : ")[0].trim(), subMethod.split(" : ")[1].trim(), returnFrom);
```

Talank commented on dt-impact-tracer/src/main/java/edu/washington/cs/dt/impact/util/RuntimeGenerator.java:
```diff
@@ -130,18 +139,84 @@ public static void main(String[] args) throws IOException {
-            bufferedWriter.close();
+    }
+
+    private static void createeElemnthead(Document doc,Element test,String test_val,String execution_time_val)
```
```suggestion
    private static void createElemnthead(Document doc,Element test,String test_val,String execution_time_val)
```
Also can you fix it where else it has been called

pioushasan2305 replied:
@Talank Noted.

pioushasan2305 committed:
Renamed temporary text file and fixed typo in function name

pioushasan2305 committed:
XML serialize deserialize via Jackson mapper,created maven surefire report in csv and compared with our result

pioushasan2305 commented:
In new commit these progress were done.
1. Changed current xml format (Need a bit more changes for required format)
2. Converted the way serialization and deserialization of XML is done to use Jackson(Done)
3. Captured all methods that are transitively invoked by a test on old version(Done)
4. Put Maven surefire runtime information into csv format(Done)
5. Compare time from surefire reports vs XML files you generated(Done)
@winglam @Talank

pioushasan2305 committed:
Fixed id format issue in xml file

pioushasan2305 committed:
method shown in linked-list type of structure

pioushasan2305 committed:
Added transitive method with fully qualified name and created tree like structure

pioushasan2305 committed:
fixed tracing bug for couting time

pioushasan2305 committed:
upgraded to version 1.2 of testrunner and changed example.sh for wildfly

pioushasan2305 committed:
updated example.sh for wildfly

pioushasan2305 committed:
Run just the one of the failing test by itself without any test and with instrumentation and testrunner

pioushasan2305 committed:
fixed instrumentation getting exception
